Platform/CMakeLists.txt

```diff
@@ -19,6 +19,25 @@ if(WIN32 AND NOT CMAKE_HOST_SYSTEM_NAME MATCHES Windows)
  return()
 endif()
 
+# With MSVC, we check that the compiler supports C++ 2011.
+# We can check the value of MSVC_VERSION with a test like this:
+#   if(MSVC_VERSION LESS 1800 OR MSVC_VERSION EQUAL 1800)
+#   endif()
+# However, there is a special version of MSVC 2013 (CTP) that supports C++ 2011
+# features, so we make a test to see if compiler supports C++ 2011.
```

Comment:
I really like having CMake check the generator (MSVC) version. My personal opinion is to uncomment the above block (and give a FATAL_ERROR if the MSVC version is too low) and comment out the block below.